diff --git a/org.eclipse.jgit.junit.http/src/org/eclipse/jgit/junit/http/AppServer.java b/org.eclipse.jgit.junit.http/src/org/eclipse/jgit/junit/http/AppServer.java
index ce04bdf..3414bc5 100644
--- a/org.eclipse.jgit.junit.http/src/org/eclipse/jgit/junit/http/AppServer.java
+++ b/org.eclipse.jgit.junit.http/src/org/eclipse/jgit/junit/http/AppServer.java
@@ -172,6 +172,16 @@ protected UserIdentity loadUser(String who) {
 			protected void loadUsers() throws IOException {
 				putUser(username, new Password(password), new String[] { role });
 			}
+
+			@Override
+			protected KnownUser loadUserInfo(String who) {
+				return null;
+			}
+
+			@Override
+			protected String[] loadRoleInfo (KnownUser who) {
+				return null;
+			}
 		};
 
 		ConstraintMapping cm = new ConstraintMapping();
