--- org.eclipse.jgit.junit.http/src/org/eclipse/jgit/junit/http/RecordingLogger.java.orig	2016-07-26 20:59:42.752829655 +0100
+++ org.eclipse.jgit.junit.http/src/org/eclipse/jgit/junit/http/RecordingLogger.java	2016-07-26 20:59:47.320778004 +0100
@@ -188,7 +188,6 @@
 		// Ignore (not relevant to test failures)
 	}
 
-	@Override
 	public void debug(String msg, long value) {
 		// Ignore (not relevant to test failures)
 	}
